src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -288,7 +288,6 @@ private static void invokeWithTimeoutWithNewExecutor(ITestNGMethod tm, Object in
       ThreadTimeoutException exception = new ThreadTimeoutException("Method "
           + tm.getQualifiedName() + "()"
           + " didn't finish within the time-out " + realTimeOut);
-      exception.setStackTrace(exec.getStackTraces()[0]);

```

Comment:
> Could add a test which throws an exception before the end of the timeout?
That scenario does not traverse this code path.  because when we throw a `ThreadTimedoutException` we explicitly check if `exec.awaitTermination(realTimeOut)` returns false and when an exception is thrown before a timeout that doesn't happen.

> I think we should keep it.
I am not able to see any scenario wherein it would be used. That was why I removed off that section.